File: Api/Boardgame/Boardgame.Domain/Entities/Queue.cs
using Boardgame.Domain.Entities.Common;

namespace Boardgame.Domain.Entities;

public class Queue
{
    public Guid Id = Guid.NewGuid();
    public string QueueNumber { get; set; } = string.Empty;
    public DateTime CreateAt { get; set; }
    public TableType TableType { get; set; } = TableType.Table;
    public int NumberOfPeople { get; set; }
    public QueueStatus Status { get; set; } = QueueStatus.InQueue;

    // Reference
    public Guid UserId { get; set; }
    public UserIdentity? User { get; set; }
    public Guid BranchId { get; set; }
    public Branch? Branch { get; set; }
    public Guid? TableId { get; set; } = null;
    public TablePlay? Table { get; set; }
}

File: frontend/src/models/data/queue.ts
export interface QueueResponse {
  Id: string;
  QueueNumber: string;
  CreateAt: string;
  TableType: number;
  NumberOfPeople: number;
  Status: number;
  UserId: string;
  BranchId: string;
  TableId: string;
}

export interface QueueRequest {
  email: string;
  tableType: number;
  numberOfPeople: number;
  branchId: string;
}

export interface QueueResponseLower {
  id: string;
  queueNumber: string;
  createAt: string;
  tableType: number;
  numberOfPeople: number;
  status: number;
  userId: string;
  branchId: string;
  tableId: string;
}

export interface QueueRequestUpdateStatus {
  queueId: string;
}

export interface QueueRequestOkSuccess {
  tableId: string;
  notificationId: string;
}
